### ISSUE TITLE
ALTER query on every start with a CURRENT_TIMESTAMP default in MySQL

### ISSUE THREAD

**Issue by Ionaru** [typeorm/typeorm#2136]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
When connecting to a MySQL database with `default: () => 'CURRENT_TIMESTAMP'` on one of the properties, the column will receive an ALTER query.

https://gist.github.com/Ionaru/6744272f96b122eda1abbdf8d46131d1

Queries that are executed when running the script:
```sql
START TRANSACTION
SELECT DATABASE() AS `db_name`
SELECT * FROM `INFORMATION_SCHEMA`.`TABLES` WHERE (`TABLE_SCHEMA` = 'web_evetrack_test_2' AND `TABLE_NAME` = 'bar')
SELECT * FROM `INFORMATION_SCHEMA`.`COLUMNS` WHERE (`TABLE_SCHEMA` = 'web_evetrack_test_2' AND `TABLE_NAME` = 'bar')
SELECT * FROM `INFORMATION_SCHEMA`.`KEY_COLUMN_USAGE` WHERE `CONSTRAINT_NAME` = 'PRIMARY' AND ((`TABLE_SCHEMA` = 'web_evetrack_test_2' AND `TABLE_NAME` = 'bar'))
SELECT `SCHEMA_NAME`, `DEFAULT_CHARACTER_SET_NAME` as `CHARSET`, `DEFAULT_COLLATION_NAME` AS `COLLATION` FROM `INFORMATION_SCHEMA`.`SCHEMATA`
SELECT `s`.* FROM `INFORMATION_SCHEMA`.`STATISTICS` `s` LEFT JOIN `INFORMATION_SCHEMA`.`REFERENTIAL_CONSTRAINTS` `rc` ON `s`.`INDEX_NAME` = `rc`.`CONSTRAINT_NAME` WHERE ((`s`.`TABLE_SCHEMA` = 'web_evetrack_test_2' AND `s`.`TABLE_NAME` = 'bar')) AND `s`.`INDEX_NAME` != 'PRIMARY' AND `rc`.`CONSTRAINT_NAME` IS NULL
SELECT `kcu`.`TABLE_SCHEMA`, `kcu`.`TABLE_NAME`, `kcu`.`CONSTRAINT_NAME`, `kcu`.`COLUMN_NAME`, `kcu`.`REFERENCED_TABLE_SCHEMA`, `kcu`.`REFERENCED_TABLE_NAME`, `kcu`.`REFERENCED_COLUMN_NAME`, `rc`.`DELETE_RULE` `ON_DELETE`, `rc`.`UPDATE_RULE` `ON_UPDATE` FROM `INFORMATION_SCHEMA`.`KEY_COLUMN_USAGE` `kcu` INNER JOIN `INFORMATION_SCHEMA`.`REFERENTIAL_CONSTRAINTS` `rc` ON `rc`.`constraint_name` = `kcu`.`constraint_name` WHERE (`kcu`.`TABLE_SCHEMA` = 'web_evetrack_test_2' AND `kcu`.`TABLE_NAME` = 'bar')
ALTER TABLE `bar` CHANGE `lastLogin` `lastLogin` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP
COMMIT
```

**Edit**
Loks like this is an issue with the Date type field as a whole, the same ALTER query behaviour is happening with this column:
```typescript
    @Column({
        nullable: true,
    })
    public lastLogin?: Date;
```
```sql
...
ALTER TABLE `bar` CHANGE `tokenExpiry` `tokenExpiry` datetime NULL
COMMIT
```

**Comment by Kononnable** [comment]
@Ionaru It's been a long time, but do you still got that error? I tried to reproduce it but it worked as expected for me.

**Comment by Ionaru** [comment]
Yes, I still have this issue, Using the reproduction gist and a MySQL database I got the `ALTER` query on a second run (first one created the table).

I'll try to give some more info on my environment:
MySQL: ` 5.7.20`
Node: `v10.13.0`

```json
"dependencies": {
  "mysql": "^2.16.0",
  "reflect-metadata": "^0.1.12",
  "ts-node": "^7.0.1",
  "typeorm": "^0.2.9",
  "typescript": "^3.2.2"
}
```

**Comment by Kononnable** [comment]
Ok, tested on a wrong branch. Should be fixed after #3262 is merged.

**Comment by irfanandratama** [comment]
still not resolved

**Comment by u7min** [comment]
still not resolved in mysql/mariaDB

**Comment by joeybrunner** [comment]
any progress on this issue? it's slowing down development considerably

**Comment by imnotjames** [comment]
If that's the case please open a new issue with more details.
